File: frontend/types/trip.ts
export type BudgetCategory = "Backpacker" | "Standard" | "Luxury";
export type TravelStyle = "Family" | "Solo" | "Couple";

export interface Trip {
  id: number;
  user_id: number;
  destination: string;
  days: number;
  budget: number;
  category: BudgetCategory;
  travel_style: TravelStyle;
  daily_budget: number;
  ai_recommendation: string | null;
}

export interface CreateTripInput {
  destination: string;
  days: number;
  budget: number;
  travel_style: TravelStyle;
}

File: backend/models/trip.py
from sqlalchemy import Column, Float, ForeignKey, Integer, String, Text
from sqlalchemy.orm import relationship

from database import Base


class Trip(Base):
    __tablename__ = "trips"

    id = Column(Integer, primary_key=True)
    destination = Column(String, nullable=False)
    days = Column(Integer, nullable=False)
    budget = Column(Float, nullable=False)
    category = Column(String, nullable=False)
    travel_style = Column(String, nullable=False, default="Solo", server_default="Solo")
    daily_budget = Column(Float, nullable=False)
    ai_recommendation = Column(Text, nullable=True)
    user_id = Column(
        Integer,
        ForeignKey("users.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )

    owner = relationship("User", back_populates="trips")
